### PR TITLE
Add database types

### LINKED ISSUE #128: support database types
Would this library be interested in supporting types relating to Moneros databases? 

I have done a draft of this [here](https://github.com/Boog900/monero-rs/tree/database2/src/database_types)

### PR DESCRIPTION
**Pull request by Boog900**
closes #128 

I have added a new feature ("database") for these types

### CODE REVIEW

**Comment by Boog900** [comment]
Hi @h4sh3d I think this is ready to be merged if there is anything you would like me to change let me know :)

File: src/lib.rs [hunk 1]
```diff
@@ -68,6 +68,8 @@ mod internal_macros;
 pub mod consensus;
 pub mod blockdata;
 pub mod cryptonote;
+//#[cfg(feature = "database")]
```
Comment (h4sh3d) [inline src/lib.rs:71]: Uncomment here

**Review by h4sh3d — CHANGES_REQUESTED** [review]
Yep, almost ready to be merged.

Can you also add a new step in ci/build job after 'Build (experimental)' for only building with database feature.

**Comment by Boog900** [comment]
Closing as I have decided to define these types outside of monero-rs

**Pull request closed without merging**